### PR TITLE
Add decorators for local class names

### LINKED ISSUE #88: @localClassName and @localClassNames decorators
Since ember-decorators added [@className and @classNames decorators](https://github.com/ember-decorators/ember-decorators/blob/68362ab0ae3c7f2b7351ec31eac0f23a5bc38077/addon/component/index.js#L43-L121), I think it's time for us to do the same.

Your original comment for reference: https://github.com/ember-decorators/ember-decorators/issues/133#issuecomment-316711907

I can try my hand at a PR, if you want. :smile:

### PR DESCRIPTION
**Pull request by buschtoens**
Closes #88.

This already kinda works, but not for the first assertion, only after the flag has been changed.

I'll take another look tomorrow.

### CODE REVIEW

**Comment by dfreeman** [comment]
@buschtoens We'll see what CI says, but I pushed an update here that I think gets everything working.

The issue I was seeing was that `@localClassName foo;` was overwriting the value passed in via `{{test-component foo=...}}` with `undefined` sometime during construction. Does this fix seem sane to you?

File: addon/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { TemplateFactory } from 'htmlbars-inline-precompile';
```
Comment (dfreeman) [inline addon/index.d.ts]: Do we need this import?

Comment (buschtoens) [reply addon/index.d.ts]: I copy-pasted from `@ember-decorators/component`, but ended up pretty much removing and rewriting everything except for this import of course 😂

**Comment by dfreeman** [comment]
@buschtoens Released in ember-css-modules@1.0.0-beta.2 🙂

**Comment by binoculars** [comment]
This looks awfully similar to https://github.com/CenterForOpenScience/ember-osf-web/blob/4675920c4d53e60c62eed7a87ea84e0f4c5ab018/app/decorators/css-modules.ts

**Comment by buschtoens** [comment]
@binoculars Absolutely! I talked to @jamescdavis on Slack prior to submitting this PR. Guess I should have actually talked to you and @chrisseto. Thank you for pointing that out.

Do you want some sort of attribution added in the code? Is your [Apache License](https://github.com/CenterForOpenScience/ember-osf-web/blob/develop/LICENSE) compatible with the [MIT License](https://github.com/salsify/ember-css-modules/blob/master/LICENSE.md)?

**Comment by binoculars** [comment]
https://opensource.stackexchange.com/questions/1713/adding-feature-to-an-apache-v2-package

Attribution is required by the license, and you're free to re-use the code (this is a derivative work). I like it in this repo anyway. More people can use it and it's less that we have to maintain.

**Comment by buschtoens** [comment]
Cool. I'll open a PR adding a comment attributing it to you folks tomorrow. Thanks. :+1:

**Pull request merged**